app/controllers/test/saml_test_controller.rb

```diff
@@ -13,66 +13,39 @@ def start
       redirect_to(request.create(test_saml_settings, {}))
     end
 
-    # rubocop:disable MethodLength
-    # TODO(sbc): Refactor to address rubocop warnings
     def decode_response
-      response = OneLogin::RubySaml::Response.new(
-        params[:SAMLResponse],
-        settings: test_saml_settings
-      )
-
-      # Ruby-saml only understands validation of Assertions. If it's
-      # a LogoutRequest, just validate the signature.
-      doc = Saml::XML::Document.parse(response.document.to_s)
-      if doc.at_xpath('/samlp:Response', samlp: Saml::XML::Namespaces::PROTOCOL)
-        is_valid = response.is_valid?
-      elsif doc.at_xpath('/samlp:LogoutResponse', samlp: Saml::XML::Namespaces::PROTOCOL)
-        begin
-          is_valid = doc.valid_signature?(saml_cert)
-        rescue
-          is_valid = false
-        end
-      end
+      res = SloResponseDecoder.new(params, test_saml_settings)
+      validity = res.valid_response? || res.valid_signature?
```

Comment:
i would expect a && here...perhaps a clarifying comment here or in the decoder class?